lib/iex/lib/iex/server.ex

```diff
@@ -43,6 +43,23 @@ defmodule IEx.Server do
   end
 
   @doc """
+  Returns the PID of the IEx evaluator process if it exists.
+  """
+  @spec evaluator :: pid | nil
+  def evaluator() do
+    case IEx.Server.whereis() do
+      nil -> nil
+      server ->
```

Comment:
Sorry, maybe I was not clear enough in my original request. It is OK to look into the process dictionary here* as long as the function that looks into the process dictionary is in the same module as the process running the server. So we can revert back to the process dictionary here since we have encapsulated it in the IEx.Server module.

*it is not really ok to use the process dictionary. the only reason we use here is to avoid deadlocks when the server, evaluator or autocomplete need to access each other while the server is executing a function on the evaluator or automcplete and vice-versa.